docs/features/interceptors.md

```diff
@@ -0,0 +1,176 @@
+
+* having the compiler drop the attributes automatically (treating them as pseudo-custom attributes, perhaps)
+* making the attribute declaration in the framework "conditional" on some debug symbol which is not usually provided.
+* deliberately keeping them there. I can't think of any good reason to do this.
+
+Perhaps the first option would be best.
+
+#### File paths
+
+File paths used in `[InterceptsLocation]` must exactly match the paths on the syntax trees they refer to by ordinal comparison. `SyntaxTree.FilePath` has already applied `/pathmap` substitution, so the paths used in the attribute will be less environment-specific in many projects.
+
+The compiler does not map `#line` directives when determining if an `[InterceptsLocation]` attribute intercepts a particular call in syntax.
+
+#### Position
+
+Line and column numbers in `[InterceptsLocation]` are 1-indexed to match existing places where source locations are displayed to the user. For example, in `Diagnostic.ToString`.
+
+The location of the call is the location of the simple name syntax which denotes the interceptable method. For example, in `app.MapGet(...)`, the name syntax for `MapGet` would be considered the location of the call. For a static method call like `System.Console.WriteLine(...)`, the name syntax for `WriteLine` is the location of the call. If we allow intercepting calls to property accessors in the future (e.g `obj.Property`), we would also be able to use the name syntax in this way.
+
+#### Attribute creation
+
+The goal of the above decisions is to make it so that when source generators are filling in `[InterceptsLocation(...)]`, they simply need to read `nameSyntax.SyntaxTree.FilePath` and `nameSyntax.GetLineSpan().Span.Start` for the exact file path and position information they need to use.
+
+We should provide samples of recommended coding patterns for generator authors to show correct usage of these, including the "translation" from 0-indexed to 1-indexed positions.
+
+### Non-invocation method usages
+
+Conversion to delegate type, address-of, etc. usages of methods cannot be intercepted.
+
+Interception can only occur for calls to ordinary member methods--not constructors, delegates, properties, local functions, etc.
+
+### Arity
+
+Interceptors cannot have type parameters or be declared in generic types at any level of nesting.
+
+### Signature matching
+
+When a call is intercepted, the interceptor and interceptable methods must meet the signature matching requirements detailed below:
+- When an interceptable instance method is compared to a classic extension method, we use the extension method in reduced form for comparison. The extension method parameter with the `this` modifier is compared to the instance method `this` parameter.
+- The returns and parameters, including the `this` parameter, must have the same ref kinds and types, except that reference types with oblivious nullability can match either annotated or unannotated reference types.
```

Comment:
Spec: I don't think nullability should be required to match. In most cases, nullability issues are treated as warnings only.